src/app/static/src/js/components/workflowWizard/workflowWizard.vue

```diff
@@ -0,0 +1,136 @@
+        <cancel-dialog @cancel="cancel" @abortCancel="abortCancel" :show-modal="showModal"/>
+    </div>
+</template>
+
+<script lang="ts">
+    import Vue from "vue"
+    import step from "../workflowWizard/step.vue"
+    import {RunWorkflowMetadata, Steps} from "../../utils/types"
+    import runWorkflowReport from "../runWorkflow/runWorkflowReport.vue"
+    import runWorkflowRun from "../runWorkflow/runWorkflowRun.vue"
+    import cancelDialog from "../runWorkflow/cancelDialog.vue"
+
+    interface Data {
+        activeStep: number
+        validStep: boolean
+        showModal: boolean
+    }
+
+    interface Methods {
+        isActive: (name: string) => boolean
+        next: (name: string) => void
+        back: (name: string) => void
+        cancel: () => void
+        confirmCancel: () => void
+        abortCancel: () => void
+        handleVisibility: (name: string) => {}
+        getCurrentIndex: (name: string) => number
+        handleStepValidity: (valid: Event) => void
+    }
+
+    interface Props {
+        runWorkflowMetadata: RunWorkflowMetadata | null
+        backButtonVisible: boolean
+        steps: Steps[]
+    }
+
+    export default Vue.extend<Data, Methods, unknown, Props>({
+        name: "workflowWizard",
+        props: {
+            runWorkflowMetadata: null,
+            backButtonVisible: {
+                type: Boolean,
+                required: false,
+                default: true
+            },
+            steps: {
+                type: [],
+                required: true
+            }
+        },
+        data(): Data {
+            return {
+                activeStep: 0,
+                validStep: false,
+                showModal: false
+            }
+        },
+        methods: {
+            handleVisibility(name) {
+                const currentIndex = this.getCurrentIndex(name)
+                if (currentIndex === this.steps.length - 1) {
+                    return {
+                        cancel: true,
+                        next: false,
+                        back: this.backButtonVisible
+                    }
+                }
+                if (currentIndex === 0) {
+                    return {
+                        cancel: true,
+                        next: true,
+                        back: false
+                    }
+                }
+                return {
+                    cancel: true,
+                    next: true,
+                    back: true
+                }
+            },
+            isActive: function (name: string) {
+                return this.getCurrentIndex(name) === this.activeStep
+            },
+            next: function (name) {
+                if (this.validStep) {
+                    if (this.steps.length - 1 === this.getCurrentIndex(name)) {
+                        this.$emit("complete", true)
+                    } else {
+                        this.activeStep = this.steps.findIndex(step => step.name === name) + 1
```

Comment:
Should `validStep` be reset to false here? It defaults to false for the first step I think.

Reply:
Ideally it should be set to false but not too sure how this will behave as it depends on how validation is implemented on each components. This has been updated with the hope that validation will be handled correctly for users , having pre-populated data or yet to be completed workflow that may have valid step even though we are setting them invalid on next event.

Reply:
I think it's OK for that to be a responsibility of the component, to always notify positively that they are valid if they are. And it's good to be consistent with the initial step.